tests-configuration-datasource/src/test/java/org/jboss/hal/testsuite/test/configuration/datasource/properties/AddModule.java

```diff
@@ -0,0 +1,280 @@
+package org.jboss.hal.testsuite.test.configuration.datasource.properties;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Joiner;
+import com.google.common.io.Files;
+import com.google.common.primitives.Bytes;
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.ServerVersion;
+import org.wildfly.extras.creaper.core.online.CliException;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+
+/**
+ * <p>Adds a module (in the JBoss Modules sense) to the module repository. The necessary directory structure
+ * will be created if needed. The JAR files specified as {@code resources} will be copied to the module directory.
+ * Unless a {@code module.xml} file was specified as an argument, it will be generated automatically.</p>
+ *
+ * <p>Example of resulting cli command:</p>
+ *
+ * <pre>
+ * module add --name=name --slot=slot --resources=jars
+ *            --resource-delimiter=char --dependencies=modules --properties=properties</p>
+ * </pre>
+ */
+public class AddModule implements OnlineCommand {
```

Comment:
Why is this class added here? Wouldn't be better to use https://github.com/wildfly-extras/creaper/blob/master/commands/src/main/java/org/wildfly/extras/creaper/commands/modules/AddModule.java from creaper dependency?

Reply:
Also the name of related commit is "Added temporary Creaper command". So maybe that there is some reasoning behind this ...

Reply:
Ok, the answer is probably `moduleRootDir` param. So what about to add PR to wildfly-extras/creaper repo and add issue(/card?) to this repo for the check of new creaper version?

Reply:
Yes, this was an attempt to fix failing test which needed the parameter. I will make a PR to Creaper and a card, thanks.

Reply:
PR created https://github.com/wildfly-extras/creaper/pull/190

Reply:
Thanks!